### PR TITLE
buildd: take sudo secure_path into account for PATH

### PR DESCRIPTION
**Pull request by cjp256**
When using sudo the PATH gets set to a cofigured secure_path, which is
not quite the same as the default found in /etc/environment. Redefine
default_command_environment() to be the combination of the two with the
secure path taking precendence.

By relying on this behavior, we can set /etc/environment and stop
setting the environment out for each command.

Signed-off-by: Chris Patterson <chris.patterson@canonical.com>

- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

File: craft_providers/bases/buildd.py [hunk 1]
```diff
@@ -35,12 +35,19 @@ def default_command_environment() -> Dict[str, Optional[str]]:
...
+    PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin"
+
+    Default /etc/sudoers found in supported Ubuntu versions:
+    PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin
 
     :returns: Dictionary of environment key/values.
     """
     return dict(
-        PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin"
+        PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin"
```
Comment (sergiusens) [inline craft_providers/bases/buildd.py:50]: I would caveat this with /snap/bin being first on the list (we do so for snapcraft)

**Review by sergiusens — APPROVED** [review]

File: craft_providers/bases/buildd.py [hunk 2]
```diff
@@ -35,12 +35,19 @@ def default_command_environment() -> Dict[str, Optional[str]]:
...
     The minimum environment for the buildd image to be configured and function
     properly.  This contains the default environment found in Ubuntu's
-    /etc/environment.
+    /etc/environment, replaced with the "secure_path" defaults used by sudo for
+    instantiating PATH.  In practice it really just means the PATH set by sudo.
+
+    Default /etc/environment found in supported Ubuntu versions:
+    PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin"
+
+    Default /etc/sudoers found in supported Ubuntu versions:
```
Comment (cmatsuoka) [inline craft_providers/bases/buildd.py:44]: Maybe "Default /etc/sudoers secure_path"?

**Review by cmatsuoka — APPROVED** [review]
Lgtm, some nitpicking in docstrings.

File: craft_providers/bases/buildd.py (hunk 1 above)
Comment (cjp256) [reply craft_providers/bases/buildd.py:50]: I specifically wanted the default to match the behavior of the base.

Snapcraft or any other application can specify their own custom environment to do whatever makes sense for it.

File: craft_providers/bases/buildd.py (hunk 2 above)
Comment (cjp256) [reply craft_providers/bases/buildd.py]: Fixed, thanks!

**Pull request merged**